### PR TITLE
[GraphBolt][CUDA] Gpu graph cache cpp

### PR DESCRIPTION
**Pull request by mfbalin**
## Description
Fixing the replace method of the GPU graph cache and additional runtime checks.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by mfbalin** [comment]
@dgl-bot

File: graphbolt/src/cuda/extension/gpu_graph_cache.cu [hunk 1]
```diff
@@ -257,35 +263,42 @@ std::tuple<torch::Tensor, std::vector<torch::Tensor>> GpuGraphCache::Replace(
               static_assert(
                   sizeof(std::byte) == 1, "Byte needs to have a size of 1.");
               auto cache_missing_dtype = torch::empty(
-                  3 * num_tensors, c10::TensorOptions()
+                  4 * num_tensors, c10::TensorOptions()
```
Comment (Rhett-Ying) [inline graphbolt/src/cuda/extension/gpu_graph_cache.cu:266]: what does `4` mean? could we avoid magic number? use constant variable instead?

Comment (mfbalin) [reply graphbolt/src/cuda/extension/gpu_graph_cache.cu:266]: This allocation is for the tuple of 2 pointers and 2 int64_t's.

Comment (mfbalin) [reply graphbolt/src/cuda/extension/gpu_graph_cache.cu:266]: a single tuple thus needs 4 int64_t sized space.

Comment (Rhett-Ying) [reply graphbolt/src/cuda/extension/gpu_graph_cache.cu:266]: please add comments for it

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**